analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -171,6 +171,15 @@ def perform_analysis(args, skip_handler, context, actions, metadata):
         config_map[ClangSA.ANALYZER_NAME].set_checker_enabled(
             ReturnValueCollector.checker_analyze)
 
+    # Statistics collector checkers must be explicitly disabled
+    # as they trash the output.
+    if "clangsa" in analyzers:
```

Comment:
If I enable the clangsa analyzer with the `--stats` flag the checkers will be disabled even if they were enabled previously.
I think we should check here if stats was enabled.
`if 'stats_enabled' in args and args.stats_enabled` we should leave them enabled.

Reply:
Collector checkers will be be enabled even if --stats are given, because they are explicitly added in pre_analaysis_manager: https://github.com/Ericsson/codechecker/blob/master/analyzer/codechecker_analyzer/pre_analysis_manager.py#L38

I tried it out and also the tests pass. So I think this is fine.

Reply:
ok, I did not remember that we explicitly enable the collector checkers later.